aten/src/ATen/native/Distributions.cpp

```diff
@@ -600,4 +576,72 @@ Tensor multinomial(
   return result;
 }
 
-}} // namespace at::native
+} // namespace at::native
+
+namespace meta {
+
+// ==================================================== Normal ========================================================
+
+TORCH_META_FUNC2(normal, float_Tensor) (
+  double mean,
+  const Tensor& std,
+  c10::optional<Generator> gen
+) {
+  set_output(std.sizes(), std.options());
+}
+
+TORCH_META_FUNC2(normal, Tensor_float) (
+  const Tensor& mean,
+  double std,
+  c10::optional<Generator> gen
+) {
+  set_output(mean.sizes(), mean.options());
+}
+
+TORCH_META_FUNC2(normal, Tensor_Tensor) (
+  Tensor const& mean,
+  Tensor const& std,
+  c10::optional<Generator> gen
+) {
+  auto shape = at::infer_size(mean.sizes(), std.sizes());
+  set_output(shape, mean.options());
+}
+
+} // namespace at::meta
+
+namespace native {
+
+// ==================================================== Normal ========================================================
+
+TORCH_IMPL_FUNC(normal_float_Tensor_out) (
+  double mean,
+  const Tensor& std,
+  c10::optional<Generator> gen,
+  const Tensor& out
+) {
+  at::native::templates::normal_out_impl<NormalStub, Generator>(
+    const_cast<Tensor&>(out), mean, std, gen);
```

Comment:
I think it's a good idea to propagate the `const` to `normal_out_impl` functions (might be BC-breaking, not sure), instead of `const_cast`-ing.

Reply:
fwiw, this just models what some other operator does already, but yeah, i agree.